### PR TITLE
Add Istio booster implementation

### PR DESCRIPTION
**Pull request by metacosm**
Also unifies wording across Spring Boot boosters.

### CODE REVIEW

**Comment by Ladicek** [comment]
I dont have any interest here, but WDYT about squashing those 20 commits to 1 or 2? :-)

File: spring-boot/current-community/istio-distributed-tracing/booster.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: Istio - Spring Boot - Distributed Tracing
+description: Booster to demonstrate how Istio's service mesh and Distributed Tracing - Jaeger interact.
+source:
```
Comment (gastaldi) [inline spring-boot/current-community/istio-distributed-tracing/booster.yaml]: No `runsOn` tag here?

File: spring-boot/current-community/istio-routing/booster.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: Istio - Spring Boot - Routing
+description: Booster to demonstrate how Istio can be used to route traffic to/from services, including load balancing multiple different versions of the same service.
+source:
```
Comment (gastaldi) [inline spring-boot/current-community/istio-routing/booster.yaml]: No `runsOn` tag here?

File: spring-boot/current-community/istio-security/booster.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+name: Istio - Spring Boot - Security
+description: Booster to demonstrate Istio Mutual TLS and ACL functionality.
```
Comment (gastaldi) [inline spring-boot/current-community/istio-security/booster.yaml]: No `runsOn` tag here?

File: spring-boot/current-community/istio-distributed-tracing/booster.yaml (hunk 1 above)
Comment (metacosm) [reply spring-boot/current-community/istio-distributed-tracing/booster.yaml]: No idea how the Istio boosters need to be qualified, actually… :(

**Comment by metacosm** [comment]
@Ladicek @gastaldi might be better to squash the commits, indeed

File: spring-boot/current-community/istio-distributed-tracing/booster.yaml (hunk 1 above)
Comment (gastaldi) [reply spring-boot/current-community/istio-distributed-tracing/booster.yaml]: They should be tested on the clusters that this booster will run: `pro`, `starter`or `osio` or simply `none` if it should be available as a zip file only.

Comment (kenfinnigan) [reply spring-boot/current-community/istio-distributed-tracing/booster.yaml]: what's the name for OCP then?

I believe the Istio boosters are only intended to run on a local OCP install

Comment (gastaldi) [reply spring-boot/current-community/istio-distributed-tracing/booster.yaml]: Ah yes, that would be `local`. I need to put that information in the README

Comment (metacosm) [reply spring-boot/current-community/istio-distributed-tracing/booster.yaml]: Just to make sure, if we only want the Istio boosters torun on local OCP install, we should have:
```yaml
metadata:
  runsOn: 
    - 'local'
```

Comment (gastaldi) [reply spring-boot/current-community/istio-distributed-tracing/booster.yaml]: Yes, that seems to be correct. Make sure to test locally before pushing ;)

**Comment by gastaldi** [comment]
@metacosm `git rebase` is your friend too :wink:

**Comment by metacosm** [comment]
@gastaldi yes, about to squash

**Comment by metacosm** [comment]
@gastaldi why do we need staging and environment sections?

File: metadata.yaml [hunk 4]
```diff
@@ -32,7 +32,7 @@ missions:
     istio: true
 - id: istio-distributed-tracing
   name: Istio - Distributed Tracing
-  description: The Istio Distributed Tracing mission demonstrates how to integrate tracing into services running on Istio.
+  description: The Istio Distributed Tracing mission demonstrates how simple distributed tracing can be integrated with an application running on an Istio-enabled cluster.
```
Comment (kenfinnigan) [inline metadata.yaml]: Have docs reviewed these text changes? @rhoads-zach @inoxx03

Comment (rhoads-zach) [reply metadata.yaml]: @kenfinnigan LGTM.

**Pull request merged**